<html><h1>Validation FAILED.</h1><p>Build Time: 6 minutes 52 seconds 826 ms</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/feaa68509b8f3c9707374cf7dd835a43ba0b02fd">feaa68509b8f3c9707374cf7dd835a43ba0b02fd</a></p><h6>Task Summary:</h6><ul><li>subrepository-compile-jdk8 -  :x:<pre><code>subrepository-compile-jdk8
   [delete] Deleting: /opt/dev/projects/github/liferay-portal-7.1.x/null48198634.properties

subrepository-compile-jdk8:
 [sync-dir] Synchronizing /opt/dev/projects/github/liferay-binaries-cache-2020/.gradle/caches/modules-2/files-2.1 into /opt/dev/projects/github/liferay-portal-7.1.x/.gradle/caches/modules-2/files-2.1
 [sync-dir] 0 files synchronized in 276ms
     [echo] ${gradle.output.error}
      [for] subrepository-compile-jdk8: The following error occurred while executing this line:
      [for] /opt/dev/projects/github/liferay-jenkins-ee/commands/build-common-acceptance-validation.xml:27: Subrepository task FAILED: The following error occurred while executing this line:
      [for] /opt/dev/projects/github/liferay-portal-7.1.x/build-test-batch.xml:5691: The following error occurred while executing this line:
      [for] /opt/dev/projects/github/liferay-portal-7.1.x/build-common.xml:859: exec returned: 1
     [echo] </code></pre></li><li>subrepository-semantic-versioning-jdk8 -  :x:<pre><code>subrepository-semantic-versioning-jdk8:
     [echo] Deleting stale baseline log files.
     [echo] Executing baseline on /opt/dev/projects/github/liferay-portal-7.1.x/modules/dxp/apps/commerce.
 [sync-dir] Synchronizing /opt/dev/projects/github/liferay-binaries-cache-2020/.gradle/caches/modules-2/files-2.1 into /opt/dev/projects/github/liferay-portal-7.1.x/.gradle/caches/modules-2/files-2.1
 [sync-dir] 0 files synchronized in 276ms
     [echo] ${gradle.output.error}
      [for] subrepository-semantic-versioning-jdk8: The following error occurred while executing this line:
      [for] /opt/dev/projects/github/liferay-jenkins-ee/commands/build-common-acceptance-validation.xml:27: Subrepository task FAILED: The following error occurred while executing this line:
      [for] /opt/dev/projects/github/liferay-portal-7.1.x/build-test-batch.xml:5759: The following error occurred while executing this line:
      [for] /opt/dev/projects/github/liferay-portal-7.1.x/build-common.xml:859: exec returned: 1
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-jenkins-ee/commands/build-common-acceptance-validation.xml:120: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-jenkins-ee/commands/build-common-acceptance-validation.xml:13: Keepgoing execution: 2 of 3 iterations failed.
[beanshell] 4323 GitHub API calls out of 5000 remain. GitHub API call limit will reset in 26 minutes 13 seconds 612 ms.
[beanshell]     https://api.github.com/repos/liferay/com-liferay-commerce/pulls/1643
[beanshell] Executing commands: cat /tmp/-1652502133.txt | mail -v -s "CI Failure has occurred" -r "ci_failure_notification" "qa-slave-verify-fail@liferay.com"
     [echo] A CI failure has occurred. QA Engineering has been notified. Please try again later or contact QA Engineering for an update.

BUILD FAILED
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-common.xml:11244: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-common-acceptance-validation.xml:127: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-common-acceptance-validation.xml:120: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-common-acceptance-validation.xml:13: Keepgoing execution: 2 of 3 iterations failed.</code></pre></li><li>central-requirements-jdk8 -  :x:<pre><code>Test failures detected. See below for details.</code></pre></li></ul><h5>For full console, click <a href="https://test-1-5.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/222//consoleText">here</a>.</h5><hr/><div><h6>Test Results:</h6><div><div><p>6 Tests Passed.<br/>1 Test Failed.</p></div></div><ol><li><div><a href="https://test-1-5.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/222//testReport/com.liferay.portal.modules/ModulesStructureTest/testScanBuildScripts">ModulesStructureTest.testScanBuildScripts</a><pre><code>junit.framework.AssertionFailedError: Incorrect "project.path.prefix" in modules/dxp/apps/commerce/gradle.properties expected:&lt;:[dxp]:apps:commerce&gt; but was:&lt;:[private]:apps:commerce&gt;
	at com.liferay.portal.modules.ModulesStructureTest._testGitRepoBuildScripts(ModulesStructureTest.java:1180)
	at com.liferay.portal.modules.ModulesStructureTest.access$200(ModulesStructureTest.java:65)
	at com.liferay.portal.modules.ModulesStructureTest$1.preVisitDirectory(ModulesStructureTest.java:159)
	at com.liferay.portal.modules.ModulesStructureTest$1.preVisitDirectory(ModulesStructureTest.java:121)
	at java.nio.file.Files.walkFileTree(Files.java:2677)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at com.liferay.portal.modules.ModulesStructureTest.testScanBuildScripts(ModulesStructureTest.java:119)
</code></pre></div></li></ol><h5>For all test results, click <a href="https://test-1-5.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/222//testReport">here</a>.</h5></div></html>

--- a/commerce-account-api/src/main/java/com/liferay/commerce/account/util/CommerceAccountHelper.java
+++ b/commerce-account-api/src/main/java/com/liferay/commerce/account/util/CommerceAccountHelper.java
@@ -42,7 +42,7 @@ public interface CommerceAccountHelper {
 			long groupId, HttpServletRequest httpServletRequest)
 		throws PortalException;
 
-	public long[] getUserCommerceAccountIds(long userId, long siteGroupId)
+	public long[] getUserCommerceAccountIds(long userId, long channelGroupId)
 		throws PortalException;
 
 	public void setCurrentCommerceAccount(

--- a/commerce-frontend-taglib/src/main/resources/META-INF/resources/mini_cart/MiniCart.es.js
+++ b/commerce-frontend-taglib/src/main/resources/META-INF/resources/mini_cart/MiniCart.es.js
@@ -380,7 +380,6 @@ class Cart extends Component {
 					this.products = updatedCart.products;
 					this.summary = updatedCart.summary;
 					this.valid = updatedCart.valid;
-					console.log(this.valid);
 					return !!(this.products && this.summary);
 				}
 			)

--- a/commerce-test/src/testIntegration/java/com/liferay/commerce/internal/order/test/CommerceOrderTest.java
+++ b/commerce-test/src/testIntegration/java/com/liferay/commerce/internal/order/test/CommerceOrderTest.java
@@ -20,15 +20,24 @@ import com.liferay.commerce.account.model.CommerceAccount;
 import com.liferay.commerce.account.service.CommerceAccountLocalService;
 import com.liferay.commerce.account.service.CommerceAccountUserRelLocalService;
 import com.liferay.commerce.account.util.CommerceAccountHelper;
+import com.liferay.commerce.constants.CommerceAddressConstants;
 import com.liferay.commerce.constants.CommerceOrderConstants;
+import com.liferay.commerce.context.CommerceContext;
 import com.liferay.commerce.currency.model.CommerceCurrency;
 import com.liferay.commerce.currency.test.util.CommerceCurrencyTestUtil;
+import com.liferay.commerce.model.CommerceAddress;
+import com.liferay.commerce.model.CommerceCountry;
 import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.model.CommerceRegion;
 import com.liferay.commerce.product.model.CommerceChannel;
 import com.liferay.commerce.product.model.CommerceChannelConstants;
 import com.liferay.commerce.product.service.CommerceChannelLocalService;
+import com.liferay.commerce.service.CommerceAddressLocalService;
+import com.liferay.commerce.service.CommerceCountryLocalService;
 import com.liferay.commerce.service.CommerceOrderLocalService;
 import com.liferay.commerce.service.CommerceOrderService;
+import com.liferay.commerce.service.CommerceRegionLocalService;
+import com.liferay.commerce.test.util.TestCommerceContext;
 import com.liferay.petra.lang.CentralizedThreadLocal;
 import com.liferay.petra.string.StringPool;
 import com.liferay.portal.kernel.dao.orm.QueryUtil;
@@ -46,6 +55,7 @@ import com.liferay.portal.kernel.test.util.GroupTestUtil;
 import com.liferay.portal.kernel.test.util.RandomTestUtil;
 import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
 import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.ListUtil;
 import com.liferay.portal.test.rule.Inject;
 import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
 import com.liferay.portal.test.rule.PermissionCheckerTestRule;
@@ -113,63 +123,6 @@ public class CommerceOrderTest {
 	}
 
 	@Test
-	public void testGetCommerceOrder() throws Exception {
-		frutillaRule.scenario(
-			"Try to get an order based on the userId, and directly based on " +
-				"the commerceAccountId"
-		).given(
-			"A B2B Site"
-		).and(
-			"A Group"
-		).and(
-			"A User"
-		).when(
-			"I try to get that order by the current user's id or the accountId"
-		).then(
-			"I should be able to get it both ways"
-		);
-
-		CommerceAccount commerceAccount =
-			_commerceAccountLocalService.addBusinessCommerceAccount(
-				"Test Business Account", 0, null, null, true, null,
-				new long[] {_user.getUserId()},
-				new String[] {_user.getEmailAddress()}, _serviceContext);
-
-		long commerceChannelGroupId = _commerceChannel.getGroupId();
-
-		_commerceOrderLocalService.addCommerceOrder(
-			_user.getUserId(), commerceChannelGroupId,
-			commerceAccount.getCommerceAccountId(),
-			_commerceCurrency.getCommerceCurrencyId());
-
-		int ordersCountByAccountId =
-			_commerceOrderService.getPendingCommerceOrdersCount(
-				commerceChannelGroupId, commerceAccount.getCommerceAccountId(),
-				StringPool.BLANK);
-
-		Assert.assertEquals(1, ordersCountByAccountId);
-
-		// This simulates using
-		// commerceOrderService.getUserPendingCommerceOrdersCount()
-		// If we used that method here it would pull accounts for the omni-admin
-
-		long[] commerceAccountIds =
-			_commerceAccountHelper.getUserCommerceAccountIds(
-				_user.getUserId(), _group.getGroupId());
-
-		long ordersCountByUser =
-			_commerceOrderLocalService.getCommerceOrdersCount(
-				_group.getCompanyId(), commerceChannelGroupId,
-				commerceAccountIds, StringPool.BLANK,
-				new int[] {CommerceOrderConstants.ORDER_STATUS_OPEN}, false);
-
-		Assert.assertEquals(1, ordersCountByUser);
-
-		_commerceOrderLocalService.deleteCommerceOrders(commerceChannelGroupId);
-		_commerceAccountLocalService.deleteCommerceAccount(commerceAccount);
-	}
-
-	@Test
 	public void testGetCommerceOrderFilteredByAccount() throws Exception {
 		frutillaRule.scenario(
 			"Ensure that users can only see orders for accounts they belong to"
@@ -215,7 +168,7 @@ public class CommerceOrderTest {
 				_commerceCurrency.getCommerceCurrencyId());
 
 		List<CommerceOrder> commerceOrders = _getUserOrders(
-			commerceChannelGroupId);
+			commerceChannelGroupId, false);
 
 		Assert.assertEquals(
 			commerceOrders.toString(), 1, commerceOrders.size());
@@ -236,7 +189,7 @@ public class CommerceOrderTest {
 			new long[] {_user.getUserId()},
 			new String[] {_user.getEmailAddress()}, null, _serviceContext);
 
-		commerceOrders = _getUserOrders(commerceChannelGroupId);
+		commerceOrders = _getUserOrders(commerceChannelGroupId, false);
 
 		Assert.assertEquals(
 			commerceOrders.toString(), 2, commerceOrders.size());
@@ -249,76 +202,6 @@ public class CommerceOrderTest {
 		_commerceAccountLocalService.deleteCommerceAccount(
 			secondCommerceAccount);
 		_userLocalService.deleteUser(secondUser);
-	}
-
-	@Test
-	public void testGetCommerceOrderForOmniAdmin() throws Exception {
-		frutillaRule.scenario(
-			"Ensure that the Omni-Admin is able to pull all orders"
-		).given(
-			"A Group"
-		).and(
-			"A User"
-		).and(
-			"A random amount of Accounts"
-		).and(
-			"a random amount of Orders for each Account"
-		).when(
-			"I get the count of all Orders the Omni-Admin can see"
-		).then(
-			"I should get the same count of Orders that were created"
-		);
-
-		long commerceChannelGroupId = _commerceChannel.getGroupId();
-
-		int ordersCreated = 0;
-
-		int accountsToCreate = RandomTestUtil.randomInt(2, 10);
-
-		List<User> randomUsers = new ArrayList<>();
-		List<CommerceAccount> randomAccounts = new ArrayList<>();
-
-		for (int i = 0; i < accountsToCreate; i++) {
-			User user = UserTestUtil.addUser(true);
-
-			randomUsers.add(user);
-
-			CommerceAccount commerceAccount =
-				_commerceAccountLocalService.addBusinessCommerceAccount(
-					"Test Generated Account " + i, 0, null, null, true, null,
-					new long[] {user.getUserId()},
-					new String[] {user.getEmailAddress()}, _serviceContext);
-
-			randomAccounts.add(commerceAccount);
-
-			int ordersToCreate = RandomTestUtil.randomInt(1, 3);
-
-			for (int j = 0; j < ordersToCreate; j++) {
-				_commerceOrderLocalService.addCommerceOrder(
-					user.getUserId(), commerceChannelGroupId,
-					commerceAccount.getCommerceAccountId(),
-					_commerceCurrency.getCommerceCurrencyId());
-			}
-
-			ordersCreated += ordersToCreate;
-		}
-
-		long ordersUserCanAccessCount =
-			_commerceOrderService.getUserPendingCommerceOrdersCount(
-				_group.getCompanyId(), commerceChannelGroupId,
-				StringPool.BLANK);
-
-		Assert.assertEquals(ordersCreated, ordersUserCanAccessCount);
-
-		_commerceOrderLocalService.deleteCommerceOrders(commerceChannelGroupId);
-
-		for (CommerceAccount commerceAccount : randomAccounts) {
-			_commerceAccountLocalService.deleteCommerceAccount(commerceAccount);
-		}
-
-		for (User user : randomUsers) {
-			_userLocalService.deleteUser(user);
-		}
 	}
 
 	@Test
@@ -376,7 +259,7 @@ public class CommerceOrderTest {
 				_commerceCurrency.getCommerceCurrencyId());
 
 		List<CommerceOrder> commerceOrders = _getUserOrders(
-			commerceChannelGroupId);
+			commerceChannelGroupId, false);
 
 		Assert.assertEquals(
 			commerceOrders.toString(), 2, commerceOrders.size());
@@ -390,7 +273,7 @@ public class CommerceOrderTest {
 			secondCommerceAccount.getCommerceAccountId(),
 			new long[] {_user.getUserId()});
 
-		commerceOrders = _getUserOrders(commerceChannelGroupId);
+		commerceOrders = _getUserOrders(commerceChannelGroupId, false);
 
 		Assert.assertEquals(
 			commerceOrders.toString(), 1, commerceOrders.size());
@@ -411,24 +294,344 @@ public class CommerceOrderTest {
 		_userLocalService.deleteUser(secondUser);
 	}
 
+	@Test
+	public void testGetCommerceOrdersForOmniAdmin() throws Exception {
+		frutillaRule.scenario(
+			"Ensure that the Omni-Admin is able to pull all orders"
+		).given(
+			"A Group"
+		).and(
+			"A User"
+		).and(
+			"A random amount of Accounts"
+		).and(
+			"a random amount of Orders for each Account"
+		).when(
+			"I get the count of all Orders the Omni-Admin can see"
+		).then(
+			"I should get the same count of Orders that were created"
+		);
+
+		long commerceChannelGroupId = _commerceChannel.getGroupId();
+
+		int ordersCreated = 0;
+
+		int accountsToCreate = RandomTestUtil.randomInt(2, 10);
+
+		List<User> randomUsers = new ArrayList<>();
+		List<CommerceAccount> randomAccounts = new ArrayList<>();
+		List<CommerceOrder> randomOrders = new ArrayList<>();
+
+		for (int i = 0; i < accountsToCreate; i++) {
+			User user = UserTestUtil.addUser(true);
+
+			randomUsers.add(user);
+
+			CommerceAccount commerceAccount =
+				_commerceAccountLocalService.addBusinessCommerceAccount(
+					"Test Generated Account " + i, 0, null, null, true, null,
+					new long[] {user.getUserId()},
+					new String[] {user.getEmailAddress()}, _serviceContext);
+
+			randomAccounts.add(commerceAccount);
+
+			int ordersToCreate = RandomTestUtil.randomInt(1, 3);
+
+			for (int j = 0; j < ordersToCreate; j++) {
+				randomOrders.add(
+					_commerceOrderLocalService.addCommerceOrder(
+						user.getUserId(), commerceChannelGroupId,
+						commerceAccount.getCommerceAccountId(),
+						_commerceCurrency.getCommerceCurrencyId()));
+			}
+
+			ordersCreated += ordersToCreate;
+		}
+
+		long ordersCount =
+			_commerceOrderService.getUserPendingCommerceOrdersCount(
+				_group.getCompanyId(), commerceChannelGroupId,
+				StringPool.BLANK);
+
+		Assert.assertEquals(
+			"Assert the Omni-admin can see the count of all pending orders",
+			ordersCreated, ordersCount);
+
+		List<CommerceOrder> commerceOrders =
+			_commerceOrderService.getUserPendingCommerceOrders(
+				_group.getCompanyId(), commerceChannelGroupId, StringPool.BLANK,
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		Assert.assertEquals(
+			"Assert the Omni-admin can see all pending orders were created",
+			ListUtil.sort(randomOrders), ListUtil.sort(commerceOrders));
+
+		// Checkout a random number of orders
+
+		CommerceContext commerceContext = new TestCommerceContext(
+			_commerceCurrency, _user, _group, null, null);
+
+		List<CommerceOrder> placedCommerceOrders = new ArrayList<>();
+
+		for (CommerceOrder commerceOrder : randomOrders) {
+			if (RandomTestUtil.randomBoolean()) {
+				CommerceAddress commerceAddress = _addAddressToAccount(
+					commerceOrder.getCommerceAccountId());
+
+				commerceOrder.setBillingAddressId(
+					commerceAddress.getCommerceAddressId());
+				commerceOrder.setShippingAddressId(
+					commerceAddress.getCommerceAddressId());
+
+				commerceOrder = _commerceOrderLocalService.updateCommerceOrder(
+					commerceOrder);
+
+				placedCommerceOrders.add(
+					_commerceOrderLocalService.checkoutCommerceOrder(
+						commerceOrder.getCommerceOrderId(), commerceContext,
+						_serviceContext));
+			}
+		}
+
+		ordersCount = _commerceOrderService.getUserPlacedCommerceOrdersCount(
+			_group.getCompanyId(), commerceChannelGroupId, StringPool.BLANK);
+
+		Assert.assertEquals(
+			"Assert the Omni-admin can see the count of all placed orders",
+			placedCommerceOrders.size(), ordersCount);
+
+		commerceOrders = _commerceOrderService.getUserPlacedCommerceOrders(
+			_group.getCompanyId(), commerceChannelGroupId, StringPool.BLANK,
+			QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		Assert.assertEquals(
+			"Assert the Omni-admin can see all placed orders",
+			ListUtil.sort(placedCommerceOrders), ListUtil.sort(commerceOrders));
+
+		_commerceOrderLocalService.deleteCommerceOrders(commerceChannelGroupId);
+
+		for (CommerceAccount commerceAccount : randomAccounts) {
+			_commerceAccountLocalService.deleteCommerceAccount(commerceAccount);
+		}
+
+		for (User user : randomUsers) {
+			_userLocalService.deleteUser(user);
+		}
+	}
+
+	@Test
+	public void testGetPendingCommerceOrder() throws Exception {
+		frutillaRule.scenario(
+			"Try to get a pending order based on the userId, and directly " +
+				"based on the commerceAccountId"
+		).given(
+			"A B2B Site"
+		).and(
+			"A Group"
+		).and(
+			"A User"
+		).when(
+			"I try to get that order by the current user's id or the accountId"
+		).then(
+			"I should be able to get it both ways"
+		);
+
+		CommerceAccount commerceAccount =
+			_commerceAccountLocalService.addBusinessCommerceAccount(
+				"Test Business Account", 0, null, null, true, null,
+				new long[] {_user.getUserId()},
+				new String[] {_user.getEmailAddress()}, _serviceContext);
+
+		long commerceChannelGroupId = _commerceChannel.getGroupId();
+
+		CommerceOrder commerceOrder =
+			_commerceOrderLocalService.addCommerceOrder(
+				_user.getUserId(), commerceChannelGroupId,
+				commerceAccount.getCommerceAccountId(),
+				_commerceCurrency.getCommerceCurrencyId());
+
+		int ordersCountByAccountId =
+			_commerceOrderService.getPendingCommerceOrdersCount(
+				commerceChannelGroupId, commerceAccount.getCommerceAccountId(),
+				StringPool.BLANK);
+
+		Assert.assertEquals(1, ordersCountByAccountId);
+
+		List<CommerceOrder> commerceOrders =
+			_commerceOrderService.getPendingCommerceOrders(
+				commerceChannelGroupId, commerceAccount.getCommerceAccountId(),
+				StringPool.BLANK, 0, 1);
+
+		CommerceOrder actualCommerceOrder = commerceOrders.get(0);
+
+		Assert.assertEquals(commerceOrder, actualCommerceOrder);
+
+		long ordersCountByUser =
+			_getUserOrdersCount(commerceChannelGroupId, false);
+
+		Assert.assertEquals(1, ordersCountByUser);
+
+		commerceOrders = _getUserOrders(commerceChannelGroupId, false);
+
+		actualCommerceOrder = commerceOrders.get(0);
+
+		Assert.assertEquals(commerceOrder, actualCommerceOrder);
+
+		_commerceOrderLocalService.deleteCommerceOrders(commerceChannelGroupId);
+		_commerceAccountLocalService.deleteCommerceAccount(commerceAccount);
+	}
+
+	@Test
+	public void testGetPlacedCommerceOrder() throws Exception {
+		frutillaRule.scenario(
+			"Try to get a placed order based on the userId, and directly " +
+				"based on the commerceAccountId"
+		).given(
+			"A B2B Site"
+		).and(
+			"A Group"
+		).and(
+			"A User"
+		).when(
+			"I try to get that order by the current user's id or the accountId"
+		).then(
+			"I should be able to get it both ways"
+		);
+
+		CommerceAccount commerceAccount =
+			_commerceAccountLocalService.addBusinessCommerceAccount(
+				"Test Business Account", 0, null, null, true, null,
+				new long[] {_user.getUserId()},
+				new String[] {_user.getEmailAddress()}, _serviceContext);
+
+		long commerceChannelGroupId = _commerceChannel.getGroupId();
+
+		CommerceOrder commerceOrder =
+			_commerceOrderLocalService.addCommerceOrder(
+				_user.getUserId(), commerceChannelGroupId,
+				commerceAccount.getCommerceAccountId(),
+				_commerceCurrency.getCommerceCurrencyId());
+
+		CommerceAddress commerceAddress = _addAddressToAccount(
+			commerceAccount.getCommerceAccountId());
+
+		commerceOrder.setBillingAddressId(
+			commerceAddress.getCommerceAddressId());
+		commerceOrder.setShippingAddressId(
+			commerceAddress.getCommerceAddressId());
+
+		commerceOrder = _commerceOrderLocalService.updateCommerceOrder(
+			commerceOrder);
+
+		CommerceContext commerceContext = new TestCommerceContext(
+			_commerceCurrency, _user, _group, commerceAccount, commerceOrder);
+
+		commerceOrder = _commerceOrderLocalService.checkoutCommerceOrder(
+			commerceOrder.getCommerceOrderId(), commerceContext,
+			_serviceContext);
+
+		int ordersCountByAccountId =
+			_commerceOrderService.getPlacedCommerceOrdersCount(
+				commerceChannelGroupId, commerceAccount.getCommerceAccountId(),
+				StringPool.BLANK);
+
+		Assert.assertEquals(1, ordersCountByAccountId);
+
+		List<CommerceOrder> commerceOrders =
+			_commerceOrderService.getPlacedCommerceOrders(
+				commerceChannelGroupId, commerceAccount.getCommerceAccountId(),
+				StringPool.BLANK, 0, 1);
+
+		CommerceOrder actualCommerceOrder = commerceOrders.get(0);
+
+		Assert.assertEquals(commerceOrder, actualCommerceOrder);
+
+		long ordersCountByUser =
+			_getUserOrdersCount(commerceChannelGroupId, true);
+
+		Assert.assertEquals(1, ordersCountByUser);
+
+		commerceOrders = _getUserOrders(commerceChannelGroupId, true);
+
+		actualCommerceOrder = commerceOrders.get(0);
+
+		Assert.assertEquals(commerceOrder, actualCommerceOrder);
+
+		_commerceOrderLocalService.deleteCommerceOrders(commerceChannelGroupId);
+		_commerceAccountLocalService.deleteCommerceAccount(commerceAccount);
+		_commerceAddressLocalService.deleteCommerceAddress(commerceAddress);
+	}
+
 	@Rule
 	public FrutillaRule frutillaRule = new FrutillaRule();
 
-	private List<CommerceOrder> _getUserOrders(long groupId) throws Exception {
+	private CommerceAddress _addAddressToAccount(long commerceAccountId)
+		throws Exception {
+
+		_commerceCountry = _commerceCountryLocalService.fetchCommerceCountry(
+			_serviceContext.getCompanyId(), 000);
+
+		if (_commerceCountry == null) {
+			_commerceCountry = _commerceCountryLocalService.addCommerceCountry(
+				RandomTestUtil.randomLocaleStringMap(), true, true, "ZZ", "ZZZ",
+				000, false, RandomTestUtil.randomDouble(), true,
+				_serviceContext);
+
+			_commerceRegion = _commerceRegionLocalService.addCommerceRegion(
+				_commerceCountry.getCommerceCountryId(),
+				RandomTestUtil.randomString(), "ZZ",
+				RandomTestUtil.randomDouble(), true, _serviceContext);
+		}
+		else {
+			_commerceRegion = _commerceRegionLocalService.getCommerceRegion(
+				_commerceCountry.getCommerceCountryId(), "ZZ");
+		}
+
+		return _commerceAddressLocalService.addCommerceAddress(
+			CommerceAccount.class.getName(), commerceAccountId,
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			String.valueOf(30133), _commerceRegion.getCommerceRegionId(),
+			_commerceCountry.getCommerceCountryId(),
+			RandomTestUtil.randomString(),
+			CommerceAddressConstants.ADDRESS_TYPE_BILLING_AND_SHIPPING,
+			_serviceContext);
+	}
+
+	private List<CommerceOrder> _getUserOrders(long groupId, boolean negate)
+		throws Exception {
 
 		// This simulates using
-		// commerceOrderService.getUserPendingCommerceOrdersCount()
+		// commerceOrderService.getUser(Pending||Placed)CommerceOrders()
 		// If we used that method here it would pull accounts for the omni-admin
 
 		long[] commerceAccountIds =
 			_commerceAccountHelper.getUserCommerceAccountIds(
-				_user.getUserId(), _group.getGroupId());
+				_user.getUserId(), groupId);
 
 		return _commerceOrderLocalService.getCommerceOrders(
 			_group.getCompanyId(), groupId, commerceAccountIds,
 			StringPool.BLANK,
-			new int[] {CommerceOrderConstants.ORDER_STATUS_OPEN}, false,
+			new int[] {CommerceOrderConstants.ORDER_STATUS_OPEN}, negate,
 			QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+	}
+
+	private long _getUserOrdersCount(long groupId, boolean negate)
+		throws Exception {
+
+		// This simulates using
+		// commerceOrderService.getUser(Pending||Placed)CommerceOrdersCount()
+		// If we used that method here it would pull accounts for the omni-admin
+
+		long[] commerceAccountIds =
+			_commerceAccountHelper.getUserCommerceAccountIds(
+				_user.getUserId(), groupId);
+
+		return _commerceOrderLocalService.getCommerceOrdersCount(
+			_group.getCompanyId(), groupId, commerceAccountIds,
+			StringPool.BLANK,
+			new int[] {CommerceOrderConstants.ORDER_STATUS_OPEN}, negate);
 	}
 
 	@Inject
@@ -441,11 +644,20 @@ public class CommerceOrderTest {
 	private CommerceAccountUserRelLocalService
 		_commerceAccountUserRelLocalService;
 
+	@Inject
+	private CommerceAddressLocalService _commerceAddressLocalService;
+
 	@DeleteAfterTestRun
 	private CommerceChannel _commerceChannel;
 
 	@Inject
 	private CommerceChannelLocalService _commerceChannelLocalService;
+
+	@DeleteAfterTestRun
+	private CommerceCountry _commerceCountry;
+
+	@Inject
+	private CommerceCountryLocalService _commerceCountryLocalService;
 
 	@DeleteAfterTestRun
 	private CommerceCurrency _commerceCurrency;
@@ -455,6 +667,12 @@ public class CommerceOrderTest {
 
 	@Inject
 	private CommerceOrderService _commerceOrderService;
+
+	@DeleteAfterTestRun
+	private CommerceRegion _commerceRegion;
+
+	@Inject
+	private CommerceRegionLocalService _commerceRegionLocalService;
 
 	@DeleteAfterTestRun
 	private Group _group;

